### PR TITLE
Exec sync

### PR DESCRIPTION
**Pull request by mrunalp**
Still need to add tests and support timeout.

### CODE REVIEW

**Comment by mrunalp** [comment]
okay, pushed commits for timeout and tests. Ready for review now.

**Comment by mrunalp** [comment]
@runcom PTAL

File: oci/oci.go [hunk 1]
```diff
@@ -148,6 +148,56 @@ func (r *Runtime) StartContainer(c *Container) error {
 	return nil
 }
 
+// ExecSync execs a command in a container and returns it's stdout, stderr and return code.
+func (r *Runtime) ExecSync(c *Container, command []string, timeout int64) (stdout []byte, stderr []byte, exitCode int32, err error) {
```
Comment (runcom) [inline oci/oci.go]: can we return a struct here with stdout, stderr, exit status?

File: oci/oci.go [hunk 2]
```diff
@@ -148,6 +148,56 @@ func (r *Runtime) StartContainer(c *Container) error {
...
+		}
+	} else {
+		err = cmd.Wait()
+		if err != nil {
+			if exitErr, ok := err.(*exec.ExitError); ok {
+				if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+					return stdoutBuf.Bytes(), stderrBuf.Bytes(), int32(status.ExitStatus()), err
+				}
+			}
+		}
```
Comment (runcom) [inline oci/oci.go]: should we return the error even if it's not an `ExitError` for debugging and understanding what happened?

File: oci/oci.go [hunk 3]
```diff
@@ -148,6 +148,56 @@ func (r *Runtime) StartContainer(c *Container) error {
...
+				return stdoutBuf.Bytes(), stderrBuf.Bytes(), -1, fmt.Errorf("failed to kill process on timeout: %v", err)
+			}
+			return stdoutBuf.Bytes(), stderrBuf.Bytes(), -1, fmt.Errorf("command timed out")
+		case err = <-done:
+			if err != nil {
+				if exitErr, ok := err.(*exec.ExitError); ok {
+					if status, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+						return stdoutBuf.Bytes(), stderrBuf.Bytes(), int32(status.ExitStatus()), err
+					}
+				}
```
Comment (runcom) [inline oci/oci.go]: should we return the error even if it's not an `ExitError` for debugging and understanding what happened?

**Comment by runcom** [comment]
lgtm except for some nits - interesting how `ExecSync` won't check the error till replying with a response though

File: oci/oci.go (hunk 2 above)
Comment (mrunalp) [inline oci/oci.go]: I missed that. Thanks for catching it :)

**Comment by mrunalp** [comment]
@runcom Pushed changes as requested.

**Comment by runcom** [comment]
LGTM will merge after tests

**Comment by runcom** [comment]
Tests somehow hung as always :(

**Comment by runcom** [comment]
Make lint failures

**Comment by mrunalp** [comment]
okay green now :)

**Pull request merged**